src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py

```diff
@@ -129,86 +167,565 @@ def __repr__(self):
+                % (FeatureFlagConstants.TO,
+                   FeatureFlagConstants.FROM,
+                   json.dumps(percentile_allocation_dict, indent=2, ensure_ascii=False))
+            )
+
+        if percentile_from >= perecentile_to:
+            raise ValidationError(
+                "Percentile allocation '%s' must be less than '%s': \n%s"
+                % (FeatureFlagConstants.FROM,
+                   FeatureFlagConstants.TO,
+                   json.dumps(percentile_allocation_dict, indent=2, ensure_ascii=False))
+            )
+
+        return cls(variant=variant, from_=percentile_from, to=perecentile_to)
+
+    def __repr__(self):
+        feature_percentile_allocation = {
+            FeatureFlagConstants.VARIANT: self.variant,
+            FeatureFlagConstants.FROM: self.From,
+            FeatureFlagConstants.TO: self.to,
+        }
+
+        return json.dumps(feature_percentile_allocation, indent=2, ensure_ascii=False)
+
+
+class FeatureAllocation:
+    """
+    Feature allocation class.
+
+    :ivar list FeatureUserAllocation user:
+        Determines how variants should be allocated for the feature to various users.
+    :ivar list FeatureGroupAllocation group:
+        Determines how variants should be allocated for the feature to various groups.
+    :ivar list FeaturePercentileAllocation percentile:
+        Determines how variants should be allocated for the feature to various users based on percentile.
+    :ivar str default_when_enabled:
+        Specifies which variant should be used
+        when the feature is considered enabled and no other allocation rules are applicable.
+    :ivar str default_when_disabled:
+        Specifies which variant should be used when the feature is considered disabled.
+    :ivar str seed:
+        The value percentile calculations are based on.
+        The calculated percentile is consistent across features for a given user if the same nonempty seed is used.
+    """
+
+    def __init__(
+        self,
+        user=None,
+        group=None,
+        percentile=None,
+        default_when_enabled=None,
+        default_when_disabled=None,
+        seed=None
+    ):
+        self.user = user
+        self.group = group
+        self.percentile = percentile
+        self.default_when_enabled = default_when_enabled
+        self.default_when_disabled = default_when_disabled
+        self.seed = seed
+
+    @classmethod
+    def convert_from_dict(cls, allocation_dict):
+        """
+        Convert JSON string to FeatureAllocation object
+
+        Args:
+            dict {string, Any}
+
+        Return:
+            FeatureAllocation object
+        """
+        default_when_disabled = allocation_dict.get(
+            FeatureFlagConstants.DEFAULT_WHEN_DISABLED, None
+        )
+        default_when_enabled = allocation_dict.get(
+            FeatureFlagConstants.DEFAULT_WHEN_ENABLED, None
+        )
+
```

Comment:
Should we also read the seed here and pass it to the initialization of the allocation object?